controllers/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -412,6 +424,115 @@ func (r *Reconciler) sendToolchainStatusNotification(logger logr.Logger,
+	result := []ComponentNotReadyStatus{}
+
+	cond, found := condition.FindConditionByType(instance.Status.Conditions, toolchainv1alpha1.ConditionReady)
+	if found && cond.Status != corev1.ConditionTrue {
+		result = append(result, ComponentNotReadyStatus{
+			ComponentType: "",
+			ComponentName: "ToolchainStatus",
+			Reason:        cond.Reason,
+			Message:       cond.Message,
+		})
+	}
+
+	if instance.Status.HostOperator != nil {
+		cond, found = condition.FindConditionByType(instance.Status.HostOperator.Conditions, toolchainv1alpha1.ConditionReady)
+		if found && cond.Status != corev1.ConditionTrue {
+			result = append(result, ComponentNotReadyStatus{
+				ComponentType: "",
+				ComponentName: "Host Operator",
+				Reason:        cond.Reason,
+				Message:       cond.Message,
+			})
+		}
+	}
+
+	if instance.Status.Members != nil {
+		for _, member := range instance.Status.Members {
+			cond, found := condition.FindConditionByType(member.MemberStatus.Conditions, toolchainv1alpha1.ConditionReady)
+			if found && cond.Status != corev1.ConditionTrue {
+				result = append(result, ComponentNotReadyStatus{
+					ComponentType: "Member",
+					ComponentName: member.ClusterName,
+					Reason:        cond.Reason,
+					Message:       cond.Message,
+				})
+			}
+
+			if member.MemberStatus.Routes != nil {
+				cond, found = condition.FindConditionByType(member.MemberStatus.Routes.Conditions, toolchainv1alpha1.ConditionReady)
+				if found && cond.Status != corev1.ConditionTrue {
+					result = append(result, ComponentNotReadyStatus{
+						ComponentType: "Member Route",
+						ComponentName: member.ClusterName,
+						Reason:        cond.Reason,
+						Message:       cond.Message,
+					})
+				}
+			}
+		}
+	}
+
+	if instance.Status.RegistrationService != nil {
+		cond, found = condition.FindConditionByType(instance.Status.RegistrationService.Deployment.Conditions, toolchainv1alpha1.ConditionReady)
+		if found && cond.Status != corev1.ConditionTrue {
+			result = append(result, ComponentNotReadyStatus{
+				ComponentType: "Registration service",
+				ComponentName: "deployment",
+				Reason:        cond.Reason,
+				Message:       cond.Message,
+			})
+		}
+
+		cond, found = condition.FindConditionByType(instance.Status.RegistrationService.Health.Conditions, toolchainv1alpha1.ConditionReady)
+		if found && cond.Status != corev1.ConditionTrue {
+			result = append(result, ComponentNotReadyStatus{
+				ComponentType: "Registration service",
+				ComponentName: "health",
+				Reason:        cond.Reason,
+				Message:       cond.Message,
+			})
+		}
+
+		cond, found = condition.FindConditionByType(instance.Status.RegistrationService.RegistrationServiceResources.Conditions, toolchainv1alpha1.ConditionReady)
+		if found && cond.Status != corev1.ConditionTrue {
+			result = append(result, ComponentNotReadyStatus{
+				ComponentType: "Registration service",
+				ComponentName: "resources",
```

Comment:
@rajivnathan can you remind me what this reg-service resources thing is?

Reply:
This is for the RegistrationService CRD that we were previously using to plug in variables when processing the Registration Service template for deployment: https://github.com/codeready-toolchain/host-operator/commit/0d131f2cef48c77b9c5040ae90571d7cb1f89fe7#diff-f93246dfeae9f687869209726c3697dce8c58e28c864b119eb86bf5bd071e8a1L91-L106

But it was removed since everything comes from the ToolchainConfig now:
https://github.com/codeready-toolchain/host-operator/commit/0d131f2cef48c77b9c5040ae90571d7cb1f89fe7#diff-dbe76ee9dfabe95c78e8d2e7c3efdf6526bfd4719168b8529c518129cb83a6acL560-L582

So everything related to the RegistrationService CRD can be removed now.

Reply:
No problem, I've removed the status check for RegistrationServiceResources.